python/GafferSceneUI/InstancerUI.py

```diff
@@ -306,6 +640,142 @@
 
 		],
 
+
+		"generateSeeds" : [
+			"description",
+			"""
+			Create a seeds context variable based on the id primvar.  This hashes the point id
+			to create a persistent integer for each instance.  The context variable is used when
+			evaluating the prototypes scene.
+			""",
+			"layout:section", "Context Variations",
+		],
+
+		"seedVariable" : [
+			"description",
+			"""
+			Name of the context variable to put the seed value in
+			""",
+			"layout:section", "Context Variations",
+			"layout:visibilityActivator", "seedsEnable",
+		],
+
+		"numSeeds" : [
+			"description",
+			"""
+			The number of possible seed values.  Increasing this allows for more different variations
+			to be driven by the seeds, increasing the total number of variations required.
+			""",
+			"layout:section", "Context Variations",
+			"layout:visibilityActivator", "seedsEnable",
+			"layout:activator", "seedsParameters",
+		],
+
+		"seedsScramble" : [
+			"description",
+			"""
+			Changing the seedsScramble changes the mapping of ids to seeds.  This results in a different
+			grouping of which instances end up with the same seed.
+			""",
+			"layout:section", "Context Variations",
+			"layout:visibilityActivator", "seedsEnable",
+			"layout:activator", "seedsParameters",
+		],
+
+		"seedsPassthroughAllIds" : [
+			"description",
+			"""
+			Enable this in rare cases it is required to pass through every single id directly into the seeds
+			context variable.  This is very expensive, because every single instance will need a separate
+			context, but is sometimes useful, and may be an acceptable cost if there isn't a huge number of
+			total instances.
+			""",
+			"label", "Passthrough All Ids",
+			"layout:section", "Context Variations",
+			"layout:visibilityActivator", "seedsEnable",
+		],
+
+		"contextVariables" : [
+			"description",
+			"""
+			Create context variables from primitive variables, these variables are put in the context when
+			evaluating the prototypes scene, to allow the prototypes scene to be generated
+			differently depending on the source point.  Supports quantization to avoid re-evaluating the
+			prototypes scene too many times.
+			""",
+			"layout:section", "Context Variations",
+			"plugValueWidget:type", "GafferSceneUI.InstancerUI._ContextVariableListWidget",
+		],
+
+		"contextVariables.*" : [
+			"deletable", True
+		],
+
+		"contextVariables.*.name" : [
+			"description",
+			"""
+			Name of the primitive variable to read.  The same name will be used for the context variables created,
+			which are used to used to evaluate the protoypes plug.
```

Comment:
Not sure where the reminder about OSL should go - there are a bunch of aspects of this where the documentation feels more general, rather than related to specific tooltips.